classes/controller/FrontController.php

```diff
@@ -1842,48 +1842,11 @@ public function getCanonicalURL()
     }
 
     /**
-     * Generate a URL corresponding to the current page but
-     * with the query string altered.
-     *
-     * If $extraParams is set to NULL, then all query params are stripped.
-     *
-     * Otherwise, params from $extraParams that have a null value are stripped,
-     * and other params are added. Params not in $extraParams are unchanged.
+     * @deprecated Since 9.0 and will be removed in the next major.
```

Comment:
Maybe we can add in the comment the new way to use this function. What do you think?

Reply:
the phpdoc in the method description should be enough